### ISSUE TITLE
Add the ability to run Flyway.clean() based on a profile's configuration

### ISSUE THREAD

**Issue by developerSid** [micronaut-projects/micronaut-configuration-flyway#2]
Would it be possible to add the ability to call the `org.flywaydb.core.Flyway.clean()` method if some kind of configuration existed for a certain profile.

My use case is one where when running tests I'd like to have Flyway clean out the schema and then migrate before the tests get run. I have an application-test.yml that is loaded when tests are run that has configuration that is slightly different from the primary application.yml such as different connection URL for the database than is normally used for development. (Basically the db is configured to run in-memory to accelerate the running of tests).

Maybe something like this (modified from `io.micronaut.configuration.dbmigration.flyway.FlywayStartupEventListener.run()`)

```java
public void run(boolean async) {
    flywayConfigurationProperties.stream()
            .filter(c -> c.isAsync() == async)
            .filter(c -> c.isCleanable()) //not sure a good way of doing this, but need to filter base on config
            .map(c -> applicationContext.findBean(Flyway.class, Qualifiers.byName(c.getNameQualifier())))
            .filter(Optional::isPresent)
            .map(Optional::get)
            .forEach(Flyway::clean);

    flywayConfigurationProperties.stream()
            .filter(c -> c.isAsync() == async)
            .map(c -> applicationContext.findBean(Flyway.class, Qualifiers.byName(c.getNameQualifier())))
            .filter(Optional::isPresent)
            .map(Optional::get)
            .forEach(Flyway::migrate);
}
```

The above example would require some configurable way to capture the desire to have the database "cleaned".  I will admit lack of knowledge as to how the Micronaut configuration system works so I'm not sure where to put something like this (or if it is even possible)

Thanks so much!

**Comment by ilopmar** [comment]
We map in Micronaut all the configuration properties that Flyway exposes in `FlywayConfigurationProperties`: https://github.com/micronaut-projects/micronaut-configuration-flyway/blob/ac1c83fd9f39312b220d04e30f5767c273a313bc/flyway/src/main/java/io/micronaut/configuration/dbmigration/flyway/FlywayConfigurationProperties.java#L46

We could use `cleanDisabled` but the problem is that, by default it is `false`, meaning that Micronaut-Flyway will always clean the database and it is the user task to set it to `true` in production to avoid the clean. I don't feel comfortable with this solution because I think it is better and safer to require user action to enable the clean of the database (even in development).
I'll discuss this internally to check for different options, so maybe we can add a new Micronaut-specific flag to control this.
P.S: I have a working version in a local branch using the flag `cleanDisabled` but I won't push it until I have the final solution.

**Comment by ilopmar** [comment]
@developerSid I've submitted a PR with the changes: https://github.com/micronaut-projects/micronaut-configuration-flyway/pull/3

**Comment by developerSid** [comment]
Awesome @ilopmar.  I'm lovin the way it looks for what it's worth.

Also sorry for being slow on the reply.

I ended up implementing your solution above for the time being with one minor addition where I implemented a truncate database "service" that gets run on `@After` which works.  Example application-test.yml
```yml
micronaut:
   environment: test
datasources:
   default:
      url: "jdbc:postgresql://localhost:6432/postgres"
      username: postgres
      password: password
      maximumPoolSize: 1
flyway:
   datasources:
      default:
         locations: classpath:db/migration/postgres
         clean-on-validation-error: true
         clean-disabled: false
```
There is one advantage to doing this in that the migration is only run when it is strictly necessary so there could be some speedup?

Having said that I do prefer your implementation (when it's released) to what I have now.

Thanks again for getting on this so fast especially this time of year!

**Comment by ilopmar** [comment]
@developerSid Thanks! If you want to try my solution you can clone the repo and build it locally. Once the PR is reviewed and merged I'll release RC2 version with the change.

**Comment by ilopmar** [comment]
@developerSid The changes are merged and available in `1.0.0.BUILD-SNAPSHOT` version. The updated documentation is here https://micronaut-projects.github.io/micronaut-configuration-flyway/snapshot/guide/index.html#configuration. Basically you can enable the cleanup using `flyway.datasources.*.clean-schema` flag.

**Comment by ilopmar** [comment]
This has been included in release `1.0.0.RC2`
